File: library/swift/src/UpstreamHttpProtocol.swift
import Foundation

/// Available upstream http protocols.
@objc
public enum UpstreamHttpProtocol: Int {
  case http1
  case http2

  /// String representation of the log level.
  var stringValue: String {
    switch self {
    case .http1:
      return "http1"
    case .http2:
      return "http2"
    }
  }
}

File: library/kotlin/src/io/envoyproxy/envoymobile/UpstreamHttpProtocol.kt
package io.envoyproxy.envoymobile

/**
 * Available upstream HTTP protocols.
 */
enum class UpstreamHttpProtocol(internal val stringValue: String) {
  HTTP1("http1"),
  HTTP2("http2");

  companion object {
    internal fun enumValue(stringRepresentation: String): UpstreamHttpProtocol {
      return when (stringRepresentation) {
        "http1" -> UpstreamHttpProtocol.HTTP1
        "http2" -> UpstreamHttpProtocol.HTTP2
        else -> throw IllegalArgumentException("Unable to find value for $stringRepresentation")
      }
    }
  }
}
